Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/MaterialBrowserInteractions.cpp

```diff
@@ -104,12 +114,50 @@ namespace AZ
                           AZ::RPI::JsonUtils::SaveObjectToFile(dialog.m_targetPath.toUtf8().constData(), materialData);
                       }
                   } });
+
+            creators.push_back(
+                { "Material_Graph_Creator", "Material Graph", QIcon(),
+                  [&](const AZStd::string& fullSourceFolderNameInCallback, [[maybe_unused]] const AZ::Uuid& sourceUUID)
+                  {
+                      const AZStd::string defaultMaterialGraphTemplate =
+                          AtomToolsFramework::GetPathWithoutAlias(AtomToolsFramework::GetSettingsValue<AZStd::string>(
+                              "/O3DE/Atom/MaterialCanvas/DefaultMaterialGraphTemplate",
+                              "@gemroot:MaterialCanvas@/Assets/MaterialCanvas/blank.materialgraphtemplate"));
+
+                      QWidget* mainWindow = nullptr;
+                      AzToolsFramework::EditorRequests::Bus::BroadcastResult(
+                          mainWindow, &AzToolsFramework::EditorRequests::Bus::Events::GetMainWindow);
+
+                      CreateDocumentDialog dialog(
+                          QObject::tr("Create Material Graph"),
+                          QObject::tr("Select Template"),
+                          QObject::tr("Select Path"),
+                          fullSourceFolderNameInCallback.c_str(),
+                          { "materialgraph" },
+                          defaultMaterialGraphTemplate.c_str(),
+                          [](const AZStd::string& path)
+                          {
+                              return IsDocumentPathEditable(path) && AZStd::wildcard_match("*.materialgraphtemplate", path);
+                          },
+                          mainWindow);
+                      dialog.adjustSize();
+
+                      if (dialog.exec() == QDialog::Accepted && !dialog.m_sourcePath.isEmpty() && !dialog.m_targetPath.isEmpty())
+                      {
+                          AZ::IO::FileIOBase::GetInstance()->Copy(
+                              dialog.m_sourcePath.toUtf8().constData(), dialog.m_targetPath.toUtf8().constData());
+                      }
+                  } });
         }
 
         bool MaterialBrowserInteractions::HandlesSource(AZStd::string_view fileName) const
         {
-            return AZStd::wildcard_match("*.material", fileName.data()) ||
-                AZStd::wildcard_match("*.materialcanvas.azasset", fileName.data());
+            return AZStd::wildcard_match("*.material", fileName) ||
+                AZStd::wildcard_match("*.materialtype", fileName) ||
+                AZStd::wildcard_match("*.materialgraph", fileName) ||
+                AZStd::wildcard_match("*.materialgraphtemplate", fileName) ||
+                AZStd::wildcard_match("*.materialgraphnode", fileName) ||
```

Comment:
Is this doing multiple searches, or  does it combine them and do 1 search?

Reply:
It's comparing one input file name to multiple extensions to see if the file name is supported by the class